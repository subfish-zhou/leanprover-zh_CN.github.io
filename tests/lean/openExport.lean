

def A.x := 10

namespace B

export A (x)

def y := 20

#check x -- works
#check y -- works

end B

#check A.x -- works
#check B.x -- works, but fails in old frontend and Lean3 :)
#check B.y -- works
#check x -- fails as expected
#check y -- fails as expected
open B
#check x -- works
#check y -- works

namespace B

#check x -- works
#check y -- works

def z := 30

#check z -- works
end B

#check z  -- works, but fails in old frontend and Lean3 :)
