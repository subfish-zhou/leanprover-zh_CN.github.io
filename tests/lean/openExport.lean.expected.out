x : Nat
y : Nat
B.x : Nat
B.x : Nat
B.y : Nat
openExport.lean:19:7-19:8: error: unknown identifier 'x'
openExport.lean:20:7-20:8: error: unknown identifier 'y'
x : Nat
y : Nat
x : Nat
y : Nat
z : Nat
z : Nat
